[WIP] voodoo/metagen command

manasdk opened the pull request:
miserable attempt at integrating metagen 

(lol .. named branch meatgen)

Commit:
voodoo/metagen command

manasdk commented:
kami - maybe there is a better way to do this but I was unable to get positional and optional args to mix with cmd2. Any idea?

manasdk commented:
The only alternative I can come up with is to completely ignore the @options and apply argparse on the input to the do_metagen method.

Commit:
Ignore @options and use argparse directly.

manasdk commented:
In 39db472adf839bba3717723e9bf4e752fb591f4c I am using argparse. Not perfect but at least takes away a ton of pain. Makes it quite tolerable.

Commit:
Use help from argparse parser.

Kami commented:
@manasdk For optional arguments you need to use `@options` decorator with cmd2.

I will look at the other changes shortly.

Kami commented on st2sdk/metagen/cmdmixin.py:
```diff
@@ -0,0 +1,120 @@
+}
+
+
+class Data(object):
```
I find this class / abstraction slightly confusing - what benefits does it provide over a dict?

manasdk replied:
Its temp. Playing around with some options without actually changing metagen. I put it in place as metagen expects an object.

Kami replied:
Ah, OK.

Overall, it looks good to me. I'm fine with merging it as soon as the build is fixed.

manasdk commented:
kami : `@options` is pretty horrible to work with. Pretty much need to duplicate help and type info. I worked around with argparse.

Kami commented:
I'm fine with using argparse as long as we use it consistency across the code base.

I know this is just a quick port so I'm fine with not refactoring any code right now, but eventually, we should build some more common building blocks so maintaining this tool and adding new commands will be easy and painless.

Commit:
Remove unnecessary code from metagen.

manasdk commented:
> I'm fine with using argparse as long as we use it consistency across the code base.

I was too optimistic about argparse and using it the way I did; it doesn't actually work. Not sure what the heck happened.

Commit:
Accept pack_path from outside of the metagen.

- bootstrap return metagen
- some adjustment to required and non-required fields
- some updates to metagen to pick up the path correctly

Kami commented:
Did you figure out the argparse stuff?

If not, I think using`sys.argv` here `data = self._get_args_parser_metagen().parse_args(args)` instead of `args` might work.

manasdk commented:
#fail
